### PR TITLE
Removing heap allocation from property flag parsing, adding some manual tests

### PR DESCRIPTION
**Pull request by jherico**

### CODE REVIEW

File: libraries/shared/src/PropertyFlags.h [hunk 1]
```diff
@@ -193,51 +195,63 @@ template<typename Enum> inline QByteArray PropertyFlags<Enum>::encode() {
...
-        char originalByte = fromEncodedBytes.at(byte);
-        for(int bit = 0; bit < BITS_PER_BYTE; bit++) {
-            int shiftBy = BITS_PER_BYTE - (bit + 1);
-            char maskBit = ( 1 << shiftBy);
-            bool bitValue = originalByte & maskBit;
-            encodedBits.setBit(byte * BITS_PER_BYTE + bit, bitValue);
+template<typename Enum> 
+inline size_t PropertyFlags<Enum>::decode(const uint8_t* data, size_t size) {
+    clear();
+    //clear(); // we are cleared out!
```
Comment (ZappoMan) [inline libraries/shared/src/PropertyFlags.h]: did you want this still?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2ce8dba8193222f6a48176e7623a5d0133c886a6 available at: https://gist.github.com/2ad109b2f0c1e1067be4<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5304.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5304.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3db3676fae7d6b794ab55b58198998a4de47facd available at: https://gist.github.com/2ad109b2f0c1e1067be4<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5304.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5304.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c1b72db18dbf750f63fb6eec86c1a455cb86ff93 available at: https://gist.github.com/2ad109b2f0c1e1067be4<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5304.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5304.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: eb9d52fbb32291d7cef7abf379b697253f57af77 available at: https://gist.github.com/2ad109b2f0c1e1067be4<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5304.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5304.dmg

**Pull request merged**